landlab/graph/sort/ext/remap_element.pyx

```diff
@@ -264,64 +263,68 @@ cdef void _offset_to_sorted_blocks(
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def offset_to_sorted_block(
-    # id_t [:, :] sorted_ids,
-    # id_t [:] offset_to_block,
-    np.ndarray[DTYPE_t, ndim=2, mode="c"] sorted_ids not None,
-    np.ndarray[DTYPE_t, ndim=1, mode="c"] offset_to_block not None,
+    id_t [:, :] sorted_ids,
+    integral_out_t [:] offset_to_block,
 ):
     cdef long n_ids = sorted_ids.shape[0]
     cdef long n_blocks = offset_to_block.shape[0]
+    cdef id_t *ptr = &sorted_ids[0, 0]
+    cdef id_t [:] sorted_ids_flat = <id_t[:sorted_ids.size]>ptr
 
     _offset_to_sorted_blocks(
-        &sorted_ids[0, 0],
+        sorted_ids_flat,
         n_ids,
         sorted_ids.shape[1],
-        &offset_to_block[0],
+        offset_to_block,
         n_blocks,
     )
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def pair_isin(
-    np.ndarray[DTYPE_t, ndim=2, mode="c"] src_pairs not None,
-    np.ndarray[DTYPE_t, ndim=2, mode="c"] pairs not None,
-    # np.ndarray[uint8, ndim=1, mode="c"] out not None,
-    np.ndarray[uint8, ndim=1, mode="c", cast=True] out not None,
+    const id_t [:, :] src_pairs,
+    const id_t [:, :] pairs,
+    uint8_t [:] out,
 ):
-    cdef long n
     cdef long pair
     cdef long n_pairs = pairs.shape[0]
     cdef long n_values = src_pairs.shape[0]
-    cdef DTYPE_t *data = <DTYPE_t *>malloc(n_values * sizeof(DTYPE_t))
+    cdef long * data_ptr = <long *>malloc(n_values * sizeof(long))
```

Comment:
Whoa. I thought python and cython did away with the need for malloc...

Reply:
I believe I could rewrite this using *numpy* arrays but I think I may get a speed hit.